python/lsst/analysis/drp/colorColorPlot.py

```diff
@@ -205,28 +211,31 @@ def colorColorPlot(self, catPlot, plotInfo):
         fig.text(0.70, 0.96, "Num. Stars: {}".format(stars.sum()), bbox=starBBox, fontsize=8)
 
         # Add colorbars
-        galCbAx = fig.add_axes([0.85, 0.11, 0.04, 0.75])
-        plt.colorbar(galPoints, cax=galCbAx, extend="both")
-        galCbAx.yaxis.set_ticks_position("left")
-        starCbAx = fig.add_axes([0.89, 0.11, 0.04, 0.75])
-        plt.colorbar(starPoints, cax=starCbAx, extend="both")
         magLabel = self.config.axisLabels["z"]
-        galText = galCbAx.text(0.5, 0.5, magLabel + ": Galaxies", color="k", rotation="vertical",
-                               transform=galCbAx.transAxes, ha="center", va="center", fontsize=10)
-        galText.set_path_effects([pathEffects.Stroke(linewidth=3, foreground="w"), pathEffects.Normal()])
-        starText = starCbAx.text(0.5, 0.5, magLabel + ": Stars", color="k", rotation="vertical",
-                                 transform=starCbAx.transAxes, ha="center", va="center", fontsize=10)
-        starText.set_path_effects([pathEffects.Stroke(linewidth=3, foreground="w"), pathEffects.Normal()])
+        if galPoints:
+            galCbAx = fig.add_axes([0.85, 0.11, 0.04, 0.75])
+            plt.colorbar(galPoints, cax=galCbAx, extend="both")
+            galCbAx.yaxis.set_ticks_position("left")
+            galText = galCbAx.text(0.5, 0.5, magLabel + ": Galaxies", color="k", rotation="vertical",
+                                   transform=galCbAx.transAxes, ha="center", va="center", fontsize=10)
+            galText.set_path_effects([pathEffects.Stroke(linewidth=3, foreground="w"), pathEffects.Normal()])
+        if starPoints:
+            starCbAx = fig.add_axes([0.89, 0.11, 0.04, 0.75])
+            plt.colorbar(starPoints, cax=starCbAx, extend="both")
+            starText = starCbAx.text(0.5, 0.5, magLabel + ": Stars", color="k", rotation="vertical",
+                                     transform=starCbAx.transAxes, ha="center", va="center", fontsize=10)
+            starText.set_path_effects([pathEffects.Stroke(linewidth=3, foreground="w"), pathEffects.Normal()])
 
         ax.set_xlabel(self.config.axisLabels["x"])
         ax.set_ylabel(self.config.axisLabels["y"])
 
         # Set useful axis limits
-        starPercsX = np.nanpercentile(xsStars, [1, 99.5])
-        starPercsY = np.nanpercentile(ysStars, [1, 99.5])
-        pad = (starPercsX[1] - starPercsX[0])/10
-        ax.set_xlim(starPercsX[0] - pad, starPercsX[1] + pad)
-        ax.set_ylim(starPercsY[0] - pad, starPercsY[1] + pad)
+        if len(xsStars) > 0:
+            starPercsX = np.nanpercentile(xsStars, [1, 99.5])
+            starPercsY = np.nanpercentile(ysStars, [1, 99.5])
+            pad = (starPercsX[1] - starPercsX[0])/10
+            ax.set_xlim(starPercsX[0] - pad, starPercsX[1] + pad)
+            ax.set_ylim(starPercsY[0] - pad, starPercsY[1] + pad)
```

Comment:
If we don't have any stars, what happens to the limits?  (Though better than crashing!)

Reply:
The example case I had had both stars and galaxies list empty. So the axis limits were [0,1] in both directions. I suppose matplotlib still autoscales the limits so they are not going to be totally useless.